internal/cmd/environment/command_update.go

```diff
@@ -34,9 +36,14 @@ func (c *command) update(cmd *cobra.Command, args []string) error {
 	}
 
 	environment := orgv2.OrgV2Environment{DisplayName: orgv2.PtrString(name)}
-	environment, err = c.V2Client.UpdateOrgEnvironment(args[0], environment)
-	if err != nil {
-		return err
+	if environment, err = c.V2Client.UpdateOrgEnvironment(args[0], environment); err != nil {
+		environmentId, err := nameconversions.EnvironmentNameToId(args[0], c.V2Client, false)
+		if err != nil {
+			return errors.NewErrorWithSuggestions(err.Error(), errors.NotValidEnvironmentIdSuggestions)
+		}
```

Comment:
I think I might be able to address this with some changes to how the conversion functions will be called. I think it would be better to instead have resource name to ID conversions embedded in our SDK wrapper functions. 

Also, for this particular case, calling `EnvironmentNameToId` function would not change anything if the user supplies a valid ID (i.e. `environmentId == args[0]` and `err == nil`) since the request is not failing because the ID is wrong, but the `UpdateOrgEnvironment` API is called a second time where it would fail again because the `--name "?"` is invalid, but it would return the backend invalid name flag error.

Reply:
Replying to the second paragraph: the issue I see with this pattern is that the code will go into the `NameToId` function even when the `id` is valid but the request is not.

If `tryGetFirst` is false, then these functions will treat the `id` as a `name`, and attempt to map it to its corresponding `id`. Since the `id` itself is not a name, it maps to nothing and we get `ResourceNameNotFoundErrorMsg`.

This causes the cli to error out before it reaches the second API call, which means it won't return the correct error.

Same thing w/ the new version of `UpdateOrgEnvironment`; the second `Execute()` is not reached in this case.

Reply:
Oh I see what you mean. When we go into the `NameToId` function for any API call we would need to try the `GET` API call to validate the ID, since the API call could fail for reasons other than an invalid ID (does this apply to `GET` calls from `describe` commands or validating flag resource IDs? or just `update` requests?) 

Do you know if the backend returns a different error code for when the resource ID is invalid vs when other parts of the API request are invalid? If that's the case we can let the second error go through and only catch the first with the `nameToId` conversion.

Reply:
> does this apply to GET calls from describe commands or validating flag resource IDs? or just update requests?
I don't think this problem applies to GET requests, since the only reason those should fail is if the `id` (or other things like the environment `id`) is invalid. Maybe a rate limit error is also possible, though?

For the second question: they should be different codes. For environment, it returns 403 unauthorized when the id is wrong and 400 when it's an otherwise bad request.

Reply:
Okay, I think I should be able to do the approach mentioned in my second question, since user flag value input is just validated with a `GET` request, so the only times `GET` requests should be failing is if the resource trying to be retrieved is an invalid ID. And for `update` requests I can check the error code and convert if it's `403`.

Reply:
We can try that, but I hesitate to filter only one one status code because the apis do have other status codes they could return, and some of them might have useful information.

Another idea: what if you build the map so that `id`s are also keys? Like:
```
name-1: [env-1],
env-1:  [env-1],
name-2: [env-2, env-3],
env-2:  [env-2],
env-3:  [env-3],
...
...
```

Reply:
I can do some more research into what codes are returned for what situations, but ideally, we would want to only call the list API for a resource if it returns an error code signifying the resource ID in the request was invalid. And hopefully (I really hope lol) that these error codes are standardized across all (or at least a majority) of our backend resources that this feature impacts.